### DISCUSSION TITLE
Auto-generate release notes comparison settings

### DISCUSSION THREAD

**Discussion by meghnabeotra** [community/community#6857] [Other Feature Feedback, Questions, & Ideas] [answered]
The `Auto-generate release notes` option is very useful to generate release notes. However, it defaults the comparison to a random release version (sometimes to a preprod release a few hours back, other times to a prod release 7 days back). I would like to always compare the 2 consecutive prod release (no matter the time span in between). Is there a pattern to it that we can work around or a setting that can be changed?

**Comment by planetf1** [comment]
I have exactly the same issue.

We do monthly releases - so I have 2.9,2.10,2.11, 3.0, 3.1, 3.2
All have a unique tag, following a consistent pattern
I'm now making 3.3, and the release notes feature is doing a comparison to 2.8 - ie from 6-7 months ago
I see now way to bound the comparison? How does the base release get set?

**Comment by planetf1** [comment]
I see no info at https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes on any kind of configuration for the baseline, so would have assumed 'previous release' . That's not what happened.

**Comment by MylesBorins** [comment]
Hey all. The intention is for it to use the previous production SemVer releases but there are a handful of edge cases and bugs in the current algorithm. Engineering is currently working on a variety of fixes that should hopefully make the experience far more reliable.

**Comment by MylesBorins** [comment, marked as answer]
Hey all. We just rolled out an update please try the feature again and please let me know if things are working again. We use an algorithm based on semver to determine prior releases, but it should work with some other versioning schemes as well

**Comment by meghnabeotra** [comment]
Thanks Myles! I tried it out last week and this week and it seems to be picking up the correct base release to compare to.

**Comment by leaanthony** [comment]
This does not appear to work for pre-releases. Example: I create a release for `v2.0.0-beta.30` and the last release was `v2.0.0-beta.29` it generates notes for commits all the way back to the last stable version, which in my case is `v1.x`. Is there a way to just provide the tags to compare rather than trying to guess? It's making big releases really hard to do.

**Comment by jhirn** [comment]
This is not working for me at all. I've tried rc and non rc tags, no matter what it's performing a diff from the selected tag `v2.53.0|v2.53.0|release_candidate_v2.53.0` against `v2.48.0` despite `v2.49.0, v2.50.0, v2.51.0,v2.52.0` all exciting.

**Comment by dylandhall** [comment]
ours invariably picks up the last tag, which means it picks up the last build - circle ci creates a tag for each build. every time i do a release it picks up the last incremental, eg `v8.1.111` => `v8.1.112`, even though the last release was `v8.0.234`. it would be great to just be able to pick which tag to use.

the generated notes are always the last issue anyone worked on and nothing else.
